src/Microsoft.Identity.Client/Internal/Cache/TokenCacheKey.cs

```diff
@@ -155,7 +156,7 @@ internal bool ScopeEquals(SortedSet<string> otherScope)
 
             if (Scope.Count == otherScope.Count)
             {
-                return this.Scope.ToLower().Intersect(otherScope.ToLower()).Count() == this.Scope.Count;
+                return this.Scope.Intersect(otherScope, StringComparer.OrdinalIgnoreCase).Count() == this.Scope.Count;
```

Comment:
This is equivalent and more roboust. Although @mjcheetham points out that scopes are case insensitive according to the spec https://tools.ietf.org/html/rfc6749#section-3.3 (please fix separately if this is a valid bug)